### ISSUE TITLE
Allow to update the delay of timers

### ISSUE THREAD

**Issue by Diarandor** [christopho/solarus#983]
If the delay in "sol.timer.start([context], delay, callback)" is given by a variable and we change its value, and then we restart the timer with "return true", the old delay is applied instead of the new value of the delay variable.

Is this intended? Or is this a bug?

EDIT: if this is intended maybe we could add new functions timer:get/set_initial_delay.

**Comment by Diarandor** [comment]
Whoops, this can be done with timer:set_remaining_time(remaining_time). Forget about this issue.

**Comment by Diarandor** [comment]
Nope, I was wrong. Restarting the timer uses the initial delay again. It would be nice to have functions:
` timer:get/set_initial_delay`

**Comment by Diarandor** [comment]
If you think this feature is not important, I can close the issue.

**Comment by llamaizing** [comment]
If I understand what you are trying to do, a better implementation might be to have it so that returning true from the callback function restarts the timer with the previous delay (current behavior), but if the callback function instead returns a number, then have the new timer restart with a delay equal to the returned number.

**Comment by Diarandor** [comment]
I like your idea of using an integer as return value to change the delay of the timer for the next loop.

**Comment by christopho** [comment]
Yes, good idea.
